src/esp32/webhook.rs

```diff
@@ -0,0 +1,192 @@
+#[allow(unused_imports)] // need this trait for `get_attribute`
+use crate::common::config::{AttributeError, Component, DynamicComponentConfig};
+use crate::proto::app::v1::RobotConfig;
+use embedded_svc::{
+    http::{
+        client::{Client as HttpClient, Connection},
+        Method,
+    },
+    io::Write,
+};
+use esp_idf_svc::http::client::{Configuration as HttpConfiguration, EspHttpConnection};
+use log::*;
+use serde_json::json;
+use thiserror::Error;
+use url::{ParseError, Url};
+
+/// Checks the board config for a `webhook` and `webhook-secret` attribute, sending a GET
+/// request with credentials for an SDK to make a connection.
+pub fn handle_webhook(config: &RobotConfig) -> Result<(), WebhookError> {
```

Comment:
could we maybe move this to the impl for webhook and have the code in `entry.rs` responsible for calling the webhook constructor and this function?

Reply:
yeah I was thinking about that as well.